base/broadcast.jl

```diff
@@ -363,4 +365,50 @@ for (sigA, sigB) in ((BitArray, BitArray),
     end
 end
 
+# additional definitions for Nullable
+# fast one & two-argument implementations
+function broadcast{T}(f, x::Nullable{T})
+    restype = promote_op(f, T)
+    if x.isnull
+        Nullable{restype}()
+    else
+        Nullable{restype}(f(x.value))
+    end
+end
+
+function broadcast(f, x::Union{Nullable, Number}, y::Union{Nullable, Number})
+    if isa(x, Number) && isa(y, Number)
+        throw(ArgumentError("broadcast is not supported on numbers"))
```

Comment:
I have no idea, but let's preserve the current behavior where `broadcast(+, 1) === 1`, and change this in another PR if needed. Mixing multiple issues is the recipe for never merging a PR.